### DISCUSSION TITLE
Question about license verification

### DISCUSSION THREAD

**Discussion by derVampir** [Labs64/NetLicensing-Community#81] [Questions] [answered]
Hello netlicensing team,
 
I have a question about license verification:

We use FLOATING licensing. In your demo programs you show license checking by LicenseeNumber. We adopted this for our solution.

Am I right in assuming that the Gateway returns a positive result if the Licensee has a valid license and the license count does not max out - meaning I just need to give the LicenseeNumber and the ProductNumber to my customers? The customer does not need an actual LicenseNumber?

Thanks in advance!

**Comment by r-brown** [comment, marked as answer]
Usually, the following entities/numbers are being used as follows:

**Product number / Product module number**
=> are delivered (hardcoded) within the product (as these hardly change)

**Licensee number**
=> need to be communicated to the customer

**([Floating](https://netlicensing.io/wiki/floating) model only) - SessionID**
=> is determined in the product and can be, for instance, serial number, IMEI number or other hardware/machine fingerprint

**License(-s)**
=> don't need to be delivered at all, as this is logic of the NetLicensing and being executed at the `validate` request
